### PR TITLE
Gy/tensor

### PR DESCRIPTION
**Pull request by pwrliang**
Implemented TensorContext based on Apache Arrow.

### CODE REVIEW

File: examples/gnn_sampler/sampler_context.h [hunk 1]
```diff
@@ -23,22 +23,21 @@ limitations under the License.
 #include <unordered_map>
 #include <vector>
 
+#include "examples/gnn_sampler/tensor_context.h"
```
Comment (yecol) [inline examples/gnn_sampler/sampler_context.h]: move to line -31

File: examples/gnn_sampler/tensor_context.h [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#ifndef EXAMPLES_GNN_SAMPLER_TENSOR_CONTEXT_H_
+#define EXAMPLES_GNN_SAMPLER_TENSOR_CONTEXT_H_
+#include "arrow/tensor.h"
```
Comment (yecol) [inline examples/gnn_sampler/tensor_context.h:18]: group the headers.

File: examples/gnn_sampler/sampler.h [hunk 3]
```diff
@@ -141,15 +141,20 @@ class Sampler : public ParallelAppBase<FRAG_T, SamplerContext<FRAG_T>>,
...
     } else {
       auto& random_result = ctx.random_result;
-      auto& ctx_data = ctx.data();
-      vertex_t v;
+      std::shared_ptr<typename context_t::arrow_tensor_t> tensor;
 
       for (auto& it : random_result) {
-        CHECK(frag.Gid2Vertex(it.first, v));
-        auto& oids = ctx_data[v];
+        if (tensor == nullptr) {
```
Comment (yecol) [inline examples/gnn_sampler/sampler.h]: move this `if` and the `data=ctx.data()` to line 145

**Review by yecol — CHANGES_REQUESTED** [review]

**Comment by pwrliang** [comment]
As we discussed before, the sampler holds nothing.

**Pull request closed without merging**